xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -217,6 +221,9 @@ public void start(Listener2 listener) {
     ldsResourceName = XdsClient.canonifyResourceName(ldsResourceName);
     callCounterProvider = SharedCallCounterMap.getInstance();
     resolveState = new ResolveState(ldsResourceName);
+    ServerInfo serverInfo = xdsClient.getServerInfo(ldsResourceName);
+    isTrustedXdsServer = serverInfo != null && serverInfo.isTrustedXdsServer();
```

Comment:
"Is trusted" should not leak out of the xds client. We want to "centrally" manage the trust. Having security checks strewn randomly through the code greatly increases the risk of a security bug.

Per the gRFC, this should only be used when validating/converting the route resources, which means `XdsRouteConfigureResource` and `XdsListenerResource`. If it isn't set in `XdsResourceType.Args.getServerInfo()`, we don't allow autoHostRewrite to be set.